pkg/cmdutil/factory.go

```diff
@@ -82,7 +86,7 @@ func executable(fallbackName string) string {
 		if p == exe {
 			return p
 		} else if f.Mode()&os.ModeSymlink != 0 {
-			if t, err := os.Readlink(p); err == nil && t == exe {
+			if t, err := filepath.EvalSymlinks(p); err == nil && t == exe {
```

Comment:
```suggestion
			realP, err := filepath.EvalSymlinks(p)
			if err != nil {
				continue
			}
			realExe, err := filepath.EvalSymlinks(exe)
			if err != nil {
				continue
			}
			if realP == realExe {
				return p
			}
```

I think this is more the logic we are looking for. Basically only use `EvalSymlinks` when necessary and verify that both `exe` and `p` are pointing to the same place.